src/containers/Layout/Layout.tsx

```diff
@@ -20,20 +19,30 @@ import MenuButton from '#components/MenuButton/MenuButton';
 import UserMenu from '#components/UserMenu/UserMenu';
 import { addQueryParam } from '#src/utils/location';
 import { getSupportedLanguages } from '#src/i18n/config';
+import { ACCESS_MODEL } from '#src/config';
 import { useProfileStore } from '#src/stores/ProfileStore';
 import { useProfiles, useSelectProfile } from '#src/hooks/useProfiles';
 import { IS_DEVELOPMENT_BUILD } from '#src/utils/common';
-import { unpersistProfile } from '#src/stores/ProfileController';
+import { getModule } from '#src/modules/container';
+import ProfileController from '#src/stores/ProfileController';
 
 const Layout = () => {
   const location = useLocation();
   const navigate = useNavigate();
   const { t, i18n } = useTranslation('common');
-  const { config, accessModel } = useConfigStore(({ config, accessModel }) => ({ config, accessModel }), shallow);
+  const { config, accessModel, getIntegration } = useConfigStore(
+    ({ config, accessModel, getIntegration }) => ({ config, accessModel, getIntegration }),
+    shallow,
+  );
+  const isLoggedIn = !!useAccountStore(({ user }) => user);
   const favoritesEnabled = !!config.features?.favoritesList;
   const { menu, assets, siteName, description, styling, features } = config;
   const metaDescription = description || t('default_description');
-  const { clientId } = useClientIntegration();
+
+  const profileController = isLoggedIn ? getModule(ProfileController) : undefined;
```

Comment:
The ternary here is making the execution more complex. Can we just add `isLoggedIn` in the if below on line 56?
```suggestion
  const profileController = getModule(ProfileController);
```